pycqed/analysis_v2/Two_state_T1_analysis.py

```diff
@@ -0,0 +1,333 @@
+            logging.warning(e)
+            self.fit_res['fit_res_P2'] = {}
+
+        doubledecay_mod = lmfit.Model(
+            DoubleExpDecayFunclocal, independent_vars='t')
+        doubledecay_mod.set_param_hint('tau1', value=10e-6, min=0, vary=True)
+        doubledecay_mod.set_param_hint(
+            'tau2', value=tau2_best, min=0, vary=False)
+        doubledecay_mod.set_param_hint(
+            'amplitude1', value=1.0, min=0, vary=True)
+        doubledecay_mod.set_param_hint(
+            'amplitude2', value=0.5, min=0, vary=True)
+        doubledecay_mod.set_param_hint('offset', value=.0, vary=True)
+
+        params2 = doubledecay_mod.make_params()
+        try:
+            for value_name in self.raw_data_dict['value_names']:
+                fit_res_P1 = doubledecay_mod.fit(
+                             data=self.proc_data_dict['P1'][value_name],
+                             t=self.proc_data_dict['time'],
+                             params=params2)
+
+            self.fit_res['fit_res_P1'] = fit_res_P1
+            tau1_best = fit_res_P1.best_values['tau1']
+            amplitude1_best = fit_res_P1.best_values['amplitude1']
+            amplitude2_best = fit_res_P1.best_values['amplitude2']
+            text_msg += ('\n' +
+                         r'$T_1^{eg}$  : ' + SI_val_to_msg_str
+                         (round(fit_res_P1.params['tau1'].value, 6), 's')[0]
+                         + SI_val_to_msg_str
+                         (fit_res_P1.params['tau1'].value, 's')[1])
+        except Exception as e:
+            logging.warning("Doulbe Fitting failed")
+            logging.warning(e)
+            self.fit_res['fit_res_P1'] = {}
+
+        doubledecay_mod = lmfit.Model(
+            DoubleExpDecayFunclocal2, independent_vars='t')
+        doubledecay_mod.set_param_hint(
+            'tau1', value=tau1_best, min=0, vary=False)
+        doubledecay_mod.set_param_hint(
+            'tau2', value=tau2_best, min=0, vary=False)
+        doubledecay_mod.set_param_hint(
+            'amplitude1', value=amplitude1_best, min=0, vary=False)
+        doubledecay_mod.set_param_hint(
+            'amplitude2', value=amplitude2_best, min=0, vary=False)
+        doubledecay_mod.set_param_hint('offset', value=.0, vary=True)
+
+        params3 = doubledecay_mod.make_params()
+        try:
+            for value_name in self.raw_data_dict['value_names']:
+                fit_res_P0 = doubledecay_mod.fit(
+                             data=self.proc_data_dict['P0'][value_name],
+                             t=self.proc_data_dict['time'],
+                             params=params3)
+
+            self.fit_res['fit_res_P0'] = fit_res_P0
+        except Exception as e:
+            logging.warning("Doulbe Fitting failed")
+            logging.warning(e)
+            self.fit_res['fit_res_P0'] = {}
+
+        self.proc_data_dict['fit_msg'] = text_msg
+
+    def prepare_plots(self):
+        val_names = self.raw_data_dict['value_names']
+
```

Comment:
Could you paste the figure that your test generates as a comment in this Pull Request? That helps assessing wether the plotting makes sense.

Reply:
![plot_populations_digitized w2_20180606_144110](https://user-images.githubusercontent.com/31408212/41499717-27745b78-7185-11e8-9dc2-1ac1a06e8e78.png)